### ISSUE TITLE
[bug] the cmake_find_package generator ignores the compiler.libcxx setting

### ISSUE THREAD

**Issue by blastrock** [conan-io/conan#10327]
### Environment Details (include every applicable attribute)
  * Operating System+version: Debian unstable
  * Compiler+version: clang 12
  * Conan version: 1.44.0
  * Python version: 3.9.9
  * CMake version: 3.22.1

### Steps to reproduce (Include if Applicable)

1. take the attached hello world project: [test.tar.gz](https://github.com/conan-io/conan/files/7856392/test.tar.gz)
2. `mkdir build`
3. `cd build`
4. `conan install .. -s compiler=clang -s compiler.version=12 -s compiler.libcxx=libc++ -s compiler.cppstd=17`
5. `conan build ..`
6. You'll see that the build.ninja file does not contain the `-stdlib=libc++` flag. You can also run `ldd test` and you'll see that it links with the default library libstdc++.so.

This setup works with the plain `cmake` generator. In that case, the generated `conanbuildinfo.cmake` contains something like

```cmake
            if(CONAN_LIBCXX STREQUAL "libstdc++" OR CONAN_LIBCXX STREQUAL "libstdc++11" )
                set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libstdc++")
            elseif(CONAN_LIBCXX STREQUAL "libc++")
                set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
            endif()
```

When using the `cmake_find_package` generator, I don't know how the flag is supposed to be forwarded. I would expect it to pass it as the `CXXFLAGS` environment variable, or the `CMAKE_CXX_FLAGS` property on the command line.

**Comment by memsharded** [comment]
Hi @blastrock 

This is expected, the ``cmake_find_package`` generator creates ``findxxxx.cmake`` files on the fly for depedencies, but it is not a generator for the current package settings. The mapping from current package settings to build configuration is implemented by the new toolchains. If you are using ``cmake_find_package`` you should maybe still be using ``cmake`` generator to get the ``libcxx`` definition.

All of this has already been implemented and released in the new CMakeDeps and CMakeToolchain integrations: https://docs.conan.io/en/latest/reference/conanfile/tools/cmake.html already available in Conan 1.44. This is the integration that will survive in Conan 2.0, all the others have already been removed in the already released 2.0-alpha releases.

**Comment by blastrock** [comment]
Hi there,

Thank you for the quick answer! I better understand how this is supposed to work.

EDIT: Removed my issue, I was using the wrong package reference ><

Thanks a lot for your help, closing this.
